pkg/app/api/insightstore/report.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2020 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package insightstore
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/pipe-cd/pipe/pkg/model"
+)
+
+// deploy frequency
+
+// deployFrequencyReport satisfy the interface `Report`
+type deployFrequencyReport struct {
+	AccumulatedTo int64                    `json:"accumulated_to"`
+	Datapoints    deployFrequencyDataPoint `json:"datapoints"`
+	FilePath      string
+}
+
+type deployFrequencyDataPoint struct {
+	Daily   map[string]deployFrequency `json:"daily"`
```

Comment:
How about instead of storing a map of `[timestamp]value`, we use a list of points (the timestamp included in the point) `Daily: []DeployFrequencyPoint`?
Not sure but I think most of the operations we want are getting the sorted list and iteration over the list so the array is more simple and better.

Reply:
Hmm.
If I use sorted list, is there any way to unmarshal directly from json with timestamp keys to list? 🤔 
Or, should we modify the json format to use a list like berow?

```
{
	"accummulated_to": 123456,
	"datapoints": {
		"daily": [
                        {
				"deploy_count": 12  // data of the first day of month
			},
			{
				"deploy_count": 13  // data of the second day of month
			}
		],
		"weekly": [{
			"deploy_count": 14
		}],
		"monthly": [{
			"deploy_count": 15
		}]
	}
}
```

Reply:
Yes, If needed we can update the JSON format as bellow:

```json
{
	"accummulated_to": 123456,
	"datapoints": {
		"daily": [
                        {
                                   "timestamp": 123456,
				"deploy_count": 12  // data of the first day of month
			},
		],
		"weekly": [
                        {
                                   "timestamp": 123456,
				"deploy_count": 12  // data of the first day of month
			},
		],
	}
}
```